frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/disks/hotplug.spec.ts

```diff
@@ -0,0 +1,271 @@
+import { DISK_SOURCE, TEMPLATE_NAME, VM_ACTION_TIMEOUT, VM_STATUS, VM_ACTION } from '../../const';
+import { ProvisionSource } from '../../enums/provisionSource';
+import { testName } from '../../support';
+import { Disk, VirtualMachineData } from '../../types/vm';
+import { confirmButton, detailViewAction } from '../../view/actions';
+import { addDisk, delDisk, waitForCurrentVMStatus } from '../../view/dialog';
+import * as tags from '../../view/selector';
+import { tab } from '../../view/tab';
+import { virtualization } from '../../view/virtualization';
+import { vm, waitForStatus } from '../../view/vm';
+
+const vmData: VirtualMachineData = {
+  name: `test-vm-hotplug-${testName.split('-')[1]}`,
+  namespace: testName,
+  template: TEMPLATE_NAME,
+  provisionSource: ProvisionSource.URL,
+  pvcSize: '1',
+  sshEnable: false,
+  startOnCreation: true,
+};
+
+const hotPlugDisk: Disk = {
+  size: '2',
+};
+
+const FCB32_URL_IMG =
+  'http://cnv-qe-server.rhevdev.lab.eng.rdu2.redhat.com/files/cnv-tests/fedora-images/Fedora-Cloud-Base-32-1.6.x86_64.qcow2';
+const FCB32_URL_REG = 'quay.io/kubevirt/fedora-cloud-container-disk-demo:latest';
+const PVC_NAME = 'hotplug-test-pvc';
+
+describe('Test UI for VM disk hot-plug', () => {
+  before(() => {
+    cy.Login();
+    cy.visit('/');
+    cy.createProject(testName);
+    vm.create(vmData);
+    waitForStatus(VM_STATUS.Running, vmData, VM_ACTION_TIMEOUT.VM_IMPORT);
+    virtualization.vms.visit();
+    cy.byLegacyTestID(vmData.name)
+      .should('exist')
+      .click();
+    tab.navigateToDisk();
+  });
+
+  after(() => {
+    cy.deleteResource({
+      kind: 'VirtualMachine',
+      metadata: {
+        name: vmData.name,
+        namespace: vmData.namespace,
+      },
+    });
+    cy.deleteResource({
+      kind: 'Namespace',
+      metadata: {
+        name: testName,
+      },
+    });
+  });
+
+  it('ID(CNV-6828) Attach Persistent hotplug disk with [Blank] as source to running VM', () => {
+    hotPlugDisk.autoDetach = false;
+    hotPlugDisk.name = 'disk-6828';
+    hotPlugDisk.source = DISK_SOURCE.Blank;
+
+    addDisk(hotPlugDisk);
+
+    cy.get(`[data-id="${hotPlugDisk.name}"]`)
+      .should('exist')
+      .should('contain', 'PersistingHotplug');
+  });
+
+  it('ID(CNV-6839) Detach hotplug disk from running VM', () => {
+    hotPlugDisk.name = 'disk-6828';
+    cy.get(`[data-id="${hotPlugDisk.name}"]`).should('exist');
+
+    delDisk(hotPlugDisk.name);
+
+    cy.get(`[data-id="${hotPlugDisk.name}"]`).should('not.exist');
+  });
+
+  it('ID(CNV-6860) Attach Persistent hotplug disk with [Import via URL] as source to running VM', () => {
+    hotPlugDisk.autoDetach = false;
```

Comment:
this line is also repeated a lot, please remove if not needed

Reply:
I'll refactor it to  two disk instances, with less code